--- sysconftool.orig	2011-11-13 05:12:47.000000000 +0400
+++ sysconftool	2012-10-09 09:16:16.000000000 +0400
@@ -26,12 +26,13 @@
 {
     my $filename=shift @ARGV;
 
-    $filename =~ s/\.dist$//;
+    my ($suffix) = $filename =~ /\.(\w+)/;
+    $filename =~ s/\.$suffix$//;
 
     my $rc;
 
     eval {
-	$rc=sysconftool($filename, $noclobber, $force);
+	$rc=sysconftool($filename, $noclobber, $force, $suffix);
     } ;
 
     if ($@)
@@ -51,16 +52,17 @@
     my $filename=shift;
     my $noclobber=shift;
     my $force=shift;
+    my $suffix=shift;
 
     my $distfile=new IO::File;
 
-    die "$filename.dist: $!\n" if ! $distfile->open("< $filename.dist");
+    die "$filename.$suffix: $!\n" if ! $distfile->open("< $filename.$suffix");
 
     my ($distheader, $distver);
 
     ($distheader, $distver)= sysconftool_readver($distfile);
 
-    die "$filename.dist: configuration header not found.\n" unless $distver;
+    die "$filename.$suffix: configuration header not found.\n" unless $distver;
 
     my $oldfile=new IO::File;
 
@@ -152,7 +154,7 @@
 	(print $newfile $distheader) || die $!;
 
 	sysconftool_writeout($newfile, $distfile, \%old_settings,
-			     \%old_version, "$filename.dist");
+			     \%old_version, "$filename.$suffix");
     } ;
 
     if ($@)
